doc/examples/example2.md

```diff
@@ -0,0 +1,67 @@
+Certification Authority Example {#example2}
+===========================================
+Certification Authority
+=======================
+
+Chain of Trust 
+--------------
+
+-Root Certificate-->Intermediate Certificate-->Client Certificate
+
+
+\code{.cpp}
+AsymmetricKeypair rootRSAKey = AsymmetricKeypair::generateRSA();
+
+DistinguishedName certDetails = DistinguishedName::Builder{}.organizationalUnitName("Car IT")
+                                                               .organizationName("BMW")
+                                                               .countryName("DE")
+                                                               .commonName("BMW internal CA Certificate").build();
+
+DistinguishedName rootCertDetails = DistinguishedName::Builder{}
+                                                                   .commonName("ImATeapot")
+                                                                   .countryName("DE")
+                                                                   .organizationName("Linux AG")
+                                                                   .organizationalUnitName("Linux Support")
+                                                                   .pkcs9EmailAddress("support@example.com")
+                                                                   .localityName("oben")
+                                                                   .stateOrProvinceName("nebenan")
+                                                                   .serialNumber("08E36DD501941432358AFE8256BC6EFD")
+                                                                   .build();
+
+BasicConstraintsExtension caConstraint{true, 1};
+
+CertificateSigningParameters caSignParams = CertificateSigningParameters::Builder{}
+            .certificateValidity(Asn1Time::Seconds(120))
+            .digestType(openssl::DigestTypes::SHA256)
+            .addExtension(caConstraint)
+            .addExtension(*_exampleUsage)
+            .build();
+
+X509Certificate rootRsaCert = CertificateAuthority::createRootCertificate(
+            rootRSAKey,
+            rootCertDetails,
+            0,
+            caSignParams);
+
+std::cout << rootRsaCert.toPEM() << std::endl;
+
+
+
+CertificateAuthority rootRsaCa = CertificateAuthority(caSignParams, 1, rootRsaCert, rootRSAKey);
+
+AsymmetricKeypair intRSAKey = AsymmetricKeypair::generateRSA();
+
+CertificateSigningRequest csr{certDetails, intRSAKey};
```

Comment:
//Create a Certificate signing request with the key for the intermediate certificates and the specific parameters of the new certificate.